src/plugins/chart_expressions/expression_metric/public/components/metric_vis.tsx

```diff
@@ -219,7 +232,33 @@ export const MetricVis = ({
       return nonNumericMetric;
     }
 
-    const baseMetric: MetricWNumber = {
+    if (isArrayOfStrings(value)) {
+      const formattedValues = value.map((v) => formatPrimaryMetric(v));
```

Comment:
I see the only difference with `MetricWText` is just the `value` type from `string` to `string[]`. It would be nice to unify these two branches.

Reply:
Yeah typescript does not like the union returns which is the reason for the many branches.

If I do something like...

```ts
if (typeof value === 'string' || isArrayOfStrings(value)) {
  const nonNumericMetric: MetricWText | MetricWStringArrayValues = {
    ...sharedMetric,
    value:
      typeof value === 'string'
        ? formatPrimaryMetric(value)
        : value.map((v) => formatPrimaryMetric(v)),
  };
  return nonNumericMetric;
}
```

...if throws an error...

```
Type '{ value: string | string[]; title: string; subtitle: string | undefined; icon: (({ width, height, color }: { width: number; height: number; color: string; }) => JSX.Element) | undefined; color: string; }' is not assignable to type 'MetricWText | MetricWStringArrayValues'.
  Type '{ value: string | string[]; title: string; subtitle: string | undefined; icon: (({ width, height, color }: { width: number; height: number; color: string; }) => JSX.Element) | undefined; color: string; }' is not assignable to type 'MetricWStringArrayValues'.
    Type '{ value: string | string[]; title: string; subtitle: string | undefined; icon: (({ width, height, color }: { width: number; height: number; color: string; }) => JSX.Element) | undefined; color: string; }' is not assignable to type '{ value: string[]; valueFormatter?: undefined; }'.
      Types of property 'value' are incompatible.
        Type 'string | string[]' is not assignable to type 'string[]'.
          Type 'string' is not assignable to type 'string[]'.ts(2322)
```